File: programmers/shim/Java/level1/20220527_약수의_개수와_덧셈.java
package Study_2022.programmers.shim.Java.level1;

class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        for(int i=left; i<=right; i++){
            if(Math.pow((int)Math.sqrt(i),2)==i)
                answer-=i;
            else answer+=i;
        }
        return answer;
    }
}

File: programmers/shim/Python/level1/20220527_약수의_개수와_덧셈.py
# import math
# def solution(left, right):
#     answer = sum(i for i in range(left, right+1) if pow(int(math.sqrt(i)),2)==i)
#     return sum(i for i in range(left, right+1))-answer*2
# 같은 계산 2번이라 비효율적일듯


import math
def solution(left, right):
    answer=0
    for i in range(left, right+1):
        if pow(int(math.sqrt(i)),2)==i:
            answer-=i
        else: answer+=i
    return answer